compiler/rustc_hir/src/target.rs

```diff
@@ -130,6 +131,29 @@ impl Target {
         }
     }
 
+    pub fn from_def_kind(def_kind: DefKind) -> Target {
```

Comment:
This is a great idea! I wonder if we should somehow merge `DefKind` and `Target` in a follow-up PR.